src/__flow-tests__/generics-tests.js

```diff
@@ -214,6 +214,131 @@ function demo_short_circuits() {
+// TODO: make a bug report :-)  (there doesn't seem to be an existing one)
+function demo_variance_short_circuit() {
+  // In addition to the perfectly sound short circuits above, Flow has
+  // another:
+  //  * If the flow is between two instances of the same generic type,
+  //    and that type has variance markings and the appropriate flows
+  //    between the respective type arguments succeed, then the overall flow
+  //    succeeds.  (The example below may make this clearer.)
+  //
+  // This would be OK so long as when the lower type (the source of the
+  // flow) in this situation is valid, that would imply the upper type (the
+  // sink of the flow) is as well.  If that were true, then:
+  //  * If the upper type is bogus, then the lower type is bogus too.  So if
+  //    the upper type gets a value at runtime through this flow, then there
+  //    was already a value in the bogus lower type.
+  //
+  // Awkwardly, that does not seem to be true.
+
+  // Define variations on our toy example above, now where the generic type
+  // is covariant or contravariant.
+  type IsStringCovar<+S: string> = string;
+  type IsStringContra<-S: string> = string;
+  // The `+` just means that IsStringCovar<S> <: IsStringCovar<T> whenever
+  // S <: T.  (When we make the definition, Flow checks this is actually
+  // true: which it is, because `string <: string` in any event.)
+  //
+  // Conversely, the `-` just means that IsStringContra<S> <: IsStringContra<T>
+  // whenever T <: S (note the swapping of sides).
+  //
+  // And now that we've declared those facts about these type aliases, and
+  // Flow has checked them, Flow is prepared to use them freely.
+
+  // Here's a flow between two equally-bogus IsStringCovar<…> types.
+  // That's fine in itself.
+  (x: IsStringCovar<3>): IsStringCovar<number> => x;
+
+  // But here's a flow from a valid IsStringCovar<…> to an invalid one.
+  // Flow still accepts it.
+  (x: IsStringCovar<string>): IsStringCovar<number | string> => x;
+
+  // And here's some perfectly runnable code, with a bogus IsStringCovar<…>
+  // on an expression that actually gets a value at runtime.
+  {
+    const f = (x: IsStringCovar<string>): IsStringCovar<number | string> => x;
+    f('a');
+    // That expression has type IsStringCovar<number | string>.
+    // Which is not a valid type, because IsStringCovar has a `string` bound
+    // on its type parameter.
+
+    // We can even have the value of bogus type get actually used:
+    console.log(String(f('a')));
+    // There's no `any` there -- flowlib has on `class String`:
+    //   static (value:mixed):string;
+    // so it takes a `mixed` and that's perfectly legit.
+    //
+    // The flows here, up to the argument to `String`, are
+    //  'a' -> IsStringCovar<string> -> IsStringCovar<number | string> -> mixed
+    // and the first is just fine; the second is accepted because variance;
+    // and the third is accepted because `mixed`.
```

Comment:
> because `mixed`

I think this is exactly the "If the flow is to `mixed`, it succeeds" short-circuit described in detail higher up in this file. I think this could be an even more emphatic reference to that—like "see `mixed` in `demo_short_circuits` above", or something. As it is, I think the connection is a little bit easy to miss. 🙂